### PR TITLE
[4.0] [com_content] Parameterized queries (backend)

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds parameterized queries to `com_content` backend.

### Testing Instructions

Test that `com_content` continues to work. Create, copy articles, change article state, feature/unfeature etc.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: administrator/components/com_content/src/Service/HTML/AdministratorService.php [hunk 1]
```diff
@@ -51,17 +51,23 @@ public function association($articleid)
...
-				->select('l.image')
-				->select('l.title as language_title');
+				->select(
+					[
+						'c.*',
+						$db->quoteName('l.sef', 'lang_sef'),
+						$db->quoteName('l.lang_code'),
+						$db->quoteName('cat.title', 'category_title'),
+						$db->quoteName('l.image'),
+						$db->quoteName('l.title as language_title'),
```
Comment (Quy) [inline administrator/components/com_content/src/Service/HTML/AdministratorService.php]:
```suggestion
						$db->quoteName('l.title', 'language_title'),
```

File: administrator/components/com_content/src/Service/HTML/AdministratorService.php [hunk 2]
```diff
@@ -51,17 +51,23 @@ public function association($articleid)
...
+					]
+				)
+				->from($db->quoteName('#__content', 'c'))
+				->join('LEFT', $db->quoteName('#__categories', 'cat'), $db->quoteName('cat.id') . ' = ' . $db->quoteName('c.catid'))
+				->join('LEFT', $db->quoteName('#__languages', 'l'), $db->quoteName('c.language') . ' = ' . $db->quoteName('l.lang_code'))
+				->whereIn($db->quoteName('c.id'), array_values($associations))
+				->where($db->quoteName('c.id') . ' != :articleId')
+				->bind(':articleId', $articleId, ParameterType::INTEGER);
```
Comment (Quy) [inline administrator/components/com_content/src/Service/HTML/AdministratorService.php]:
```suggestion
				->bind(':articleId', $articleid, ParameterType::INTEGER);
```

**Comment by Quy** [comment]
Click `Featured` icon.

> You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '?,?,?,?' at line 1

**Comment by SharkyKZ** [comment]
Thanks, should be fixed.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on bac02f48c0d4fb87a46814d56ae9d4ef055afc91<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27970">issues.joomla.org/tracker/joomla-cms/27970</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bac02f48c0d4fb87a46814d56ae9d4ef055afc91<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27970">issues.joomla.org/tracker/joomla-cms/27970</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27970">issues.joomla.org/tracker/joomla-cms/27970</a>.</sub>

**Comment by richard67** [comment]
Thanks.

**Pull request merged**